Always require freeform problem description

evilstreak opened the pull request:
Previously a problem that we had an SOR code for had an optional freeform description. To try to get all the information a DLO operative needs to do job we're changing it to require the description whether we have an SOR code or not.

See also: #96

evilstreak committed:
Always require freeform problem description

Previously a problem that we had an SOR code for had an optional
freeform description. To try to get all the information a DLO operative
needs to do job we're changing it to require the description whether we
have an SOR code or not.

leeky reviewed (approved):
LGTM! :shipit:

Although it's no longer used, it might be worth keeping in the optional field code for SimpleForm. It was a bit of a faff to get working.

evilstreak commented:
To summarise an IRL discussion: I agree on the optional field thing. The reason I deleted it was because of the 98% code coverage restriction. I'm going to drop it to 95% and then get rid of the commit that deletes the optional field.

evilstreak committed:
Drop minimum test coverage to 95%

Making the description for a diagnosed repair mandatory means we don't
have any optional fields. That means some of the code in the optional
fields extension doesn't get tested any more, dropping us below the 98%
minimum. We're not keen to remove the extension yet, nor do we want to
add a bunch of extra tests just for it, so to facilitate just shipping
this thing we're dropping the coverage %.
